### PR TITLE
Duplicate argv in TestCHIPArgParser

### PR DESCRIPTION
**Pull request by mspang**
This is necessary to make TestCHIPArgParser const correct, since it uses
string constants and getopt_long() requires mutable strings.

### CODE REVIEW

**Comment by rwalker-apple** [comment]
what version of getopt_long() requires mutable strings?
```
ETOPT_LONG(3)           BSD Library Functions Manual           GETOPT_LONG(3)

NAME
     getopt_long, getopt_long_only -- get long options from command line argument list

LIBRARY
     Standard C Library (libc, -lc)

SYNOPSIS
     #include <getopt.h>

     extern char *optarg;
     extern int optind;
     extern int optopt;
     extern int opterr;
     extern int optreset;

     int
     getopt_long(int argc, char * const *argv, const char *optstring, const struct option *longopts,
         int *longindex);

     int
     getopt_long_only(int argc, char * const *argv, const char *optstring, const struct option *longopts,
         int *longindex);

```

**Comment by rwalker-apple** [comment]
from linux:

```GETOPT(3)                  Linux Programmer's Manual                 GETOPT(3)

NAME
       getopt, getopt_long, getopt_long_only, optarg, optind, opterr, optopt -
       Parse command-line options

SYNOPSIS
       #include <unistd.h>

       int getopt(int argc, char * const argv[],
                  const char *optstring);

       extern char *optarg;
       extern int optind, opterr, optopt;

       #include <getopt.h>

       int getopt_long(int argc, char * const argv[],
                  const char *optstring,
                  const struct option *longopts, int *longindex);

       int getopt_long_only(int argc, char * const argv[],
                  const char *optstring,
                  const struct option *longopts, int *longindex);

   Feature Test Macro Requirements for glibc (see feature_test_macros(7)):

       getopt(): _POSIX_C_SOURCE >= 2 || _XOPEN_SOURCE
       getopt_long(), getopt_long_only(): _GNU_SOURCE
```

**Comment by mspang** [comment]
> what version of getopt_long() requires mutable strings?
> 

All of them. We need the declaration to be

const char * const *

to be able to use const array of string constants.

> ```
> ETOPT_LONG(3)           BSD Library Functions Manual           GETOPT_LONG(3)
> 
> NAME
>      getopt_long, getopt_long_only -- get long options from command line argument list
> 
> LIBRARY
>      Standard C Library (libc, -lc)
> 
> SYNOPSIS
>      #include <getopt.h>
> 
>      extern char *optarg;
>      extern int optind;
>      extern int optopt;
>      extern int opterr;
>      extern int optreset;
> 
>      int
>      getopt_long(int argc, char * const *argv, const char *optstring, const struct option *longopts,
>          int *longindex);
> 
>      int
>      getopt_long_only(int argc, char * const *argv, const char *optstring, const struct option *longopts,
>          int *longindex);
> ```

File: src/lib/support/tests/TestCHIPArgParserDriver.cpp [hunk 1]
```diff
@@ -25,7 +25,11 @@
 
 #include "TestSupport.h"
 
+#include <support/CHIPMem.h>
+
 int main()
 {
+    chip::Platform::MemoryInit();
+
     return (TestCHIPArgParser());
```
Comment (andy31415) [inline src/lib/support/tests/TestCHIPArgParserDriver.cpp:34]: supernit: while we touch this code, we can remove the extra brackets.

**Review by andy31415 — APPROVED** [review]

**Review by bzbarsky-apple — APPROVED** [review]
Thank you!

**Review by rwalker-apple — APPROVED** [review]

**Review by BroderickCarlin — APPROVED** [review]

**Comment by BroderickCarlin** [comment]
@mspang can you fix the merge conflict?

**Pull request merged**